test/e2e/set/scyllacluster/scyllacluster_auth.go

```diff
@@ -0,0 +1,159 @@
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = g.Describe("ScyllaCluster authentication", func() {
+	defer g.GinkgoRecover()
+
+	f := framework.NewFramework("scyllacluster")
+
+	g.It("agent requires authentication", func() {
+		ctx, cancel := context.WithTimeout(context.Background(), testTimout)
+		defer cancel()
+
+		sc := scyllaclusterfixture.BasicScyllaCluster.ReadOrFail()
+		sc.Spec.Datacenter.Racks[0].Members = 1
+
+		framework.By("Creating a ScyllaCluster")
+		err := framework.SetupScyllaClusterSA(ctx, f.KubeClient().CoreV1(), f.KubeClient().RbacV1(), f.Namespace(), sc.Name)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		sc, err = f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Create(ctx, sc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaCluster to deploy")
+		waitCtx1, waitCtx1Cancel := contextForRollout(ctx, sc)
+		defer waitCtx1Cancel()
+		sc, err = waitForScyllaClusterState(waitCtx1, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, scyllaClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		verifyScyllaCluster(ctx, f.KubeClient(), sc)
+
+		framework.By("Rejecting an unauthorized request")
+
+		_, hosts, err := getScyllaClient(ctx, f.KubeClient().CoreV1(), sc)
+
+		emptyAuthToken := ""
+		_, err = getScyllaClientStatus(ctx, hosts, emptyAuthToken)
+		o.Expect(err).To(o.HaveOccurred())
+		o.Expect(scyllaclient.StatusCodeOf(err)).To(o.Equal(http.StatusUnauthorized))
+
+		framework.By("Accepting requests authorized using token from provisioned secret")
+
+		token, err := helpers.GetAgentAuthToken(ctx, f.KubeClient().CoreV1(), sc.Name, sc.Namespace)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		_, err = getScyllaClientStatus(ctx, hosts, token)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Specifying auth token in agent config")
+
+		agentConfig := struct {
+			AuthToken string `yaml:"auth_token"`
+		}{
+			AuthToken: "42",
+		}
+		agentConfigData, err := yaml.Marshal(agentConfig)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		agentConfigSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "agent-config",
+				Namespace: f.Namespace(),
+			},
+			Type: corev1.SecretTypeOpaque,
+			Data: map[string][]byte{
+				naming.ScyllaAgentConfigFileName: agentConfigData,
+			},
+		}
+
+		_, err = f.KubeClient().CoreV1().Secrets(f.Namespace()).Create(ctx, agentConfigSecret, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		_, err = f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Patch(
+			ctx,
+			sc.Name,
+			types.MergePatchType,
+			[]byte(fmt.Sprintf(
+				`{"metadata":{"uid":"%s"},"spec":{"datacenter":{"racks":[{"name":"%s","scyllaAgentConfig":"%s","members":1,"resources":{"limits":{"cpu":"1","memory":"1Gi"},"requests":{"cpu":"10m","memory":"100Mi"}},"storage":{"capacity":"100Mi"}}]}}}`,
```

Comment:
Why are you patching the resources with the agent config?

Reply:
Without full rack spec i get:
`admission webhook "webhook.scylla.scylladb.com" denied the request: set cpu, memory resource limits for rack us-east-1a`
Looks like patch overwrites entire rack.

Reply:
good idea! done